src/main/java/org/reactome/release/Resource.java

```diff
@@ -0,0 +1,166 @@
+		checkMandatoryAttributesExist(this.resourceAsJson.toString());
+	}
+
+	private JsonObject convertCSVToJson(CSVRecord resourceInfoCSV) {
+		String resourceInfoJson = convertCSVToJsonString(resourceInfoCSV);
+
+		return new JsonParser().parse(resourceInfoJson).getAsJsonObject();
+	}
+
+	private String convertCSVToJsonString(CSVRecord resourceInfoCSV) {
+		String resourceInfoJson = "";
+		try {
+			ObjectMapper mapper = new ObjectMapper();
+			mapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+			resourceInfoJson = mapper.writeValueAsString(resourceInfoCSV.toMap());
+		} catch (JsonProcessingException e) {
+			logger.error("Unable to parse the CSV record's header to value map into JSON String", e);
+		}
+
+		return resourceInfoJson;
+	}
+
+	private void checkMandatoryAttributesExist(String originalRecord) throws IllegalArgumentException {
+		Map<String, String> mandatoryAttributesMap = new HashMap<>();
+
+		mandatoryAttributesMap.put("Release Step", getReleaseStep());
+		mandatoryAttributesMap.put("Main Program", getMainProgramName());
+		mandatoryAttributesMap.put("Dependency in Source Code", getSourceCodeDependency());
+		mandatoryAttributesMap.put("Resource", getResourceName());
+		mandatoryAttributesMap.put("Resource Description", getResourceDescription());
+		mandatoryAttributesMap.put("Resource Type", getResourceType().toString());
+		mandatoryAttributesMap.put("Resource URL", getResourceURL().toString());
+
+		for (Entry<String, String> mandatoryAttributeEntry: mandatoryAttributesMap.entrySet()) {
+			String mandatoryAttributeName = mandatoryAttributeEntry.getKey();
+			String mandatoryAttributeValue = mandatoryAttributeEntry.getValue();
+
+			if (mandatoryAttributeValue == null || mandatoryAttributeValue.isEmpty()) {
+				throw new IllegalStateException(
+					mandatoryAttributeName + " is empty or does not exist for " + originalRecord
+				);
+			}
+		}
+	}
+
+	public String getReleaseStep() {
+		return this.resourceAsJson.get("Release Step").getAsString();
+	}
+
+	public String getMainProgramName() {
+		return this.resourceAsJson.get("Main Program").getAsString();
+	}
+
+	public String getSourceCodeDependency() {
+		return this.resourceAsJson.get("Dependency in Source Code").getAsString();
+	}
+
+	public String getResourceName() {
+		return this.resourceAsJson.get("Resource").getAsString();
+	}
+
+	public String getResourceDescription() {
+		return this.resourceAsJson.get("Resource Description").getAsString();
+	}
+
+	public ResourceType getResourceType() {
+		String resourceType = this.resourceAsJson.get("Resource Type").getAsString().toUpperCase().replace(" ", "_");
+		return ResourceType.valueOf(resourceType);
+	}
+
+	public JsonObject getResourceAsJsonObject() {
+		return this.resourceAsJson;
+	}
+
+//	public String getHeaderAsTSVString() {
+//		return String.join("\t", getHeaderNameToValueMap().keySet());
+//	}
+//
+//	public String getValuesAsTSVString() {
+//		return String.join("\t", getHeaderNameToValueMap().values());
+//	}
```

Comment:
WIP?

Reply:
No, actually this was when the output was going to be in CSV format.  I'll remove it.